src/GraphQL/Validation/FieldArgumentsValidationContext.cs

```diff
@@ -0,0 +1,73 @@
+using GraphQL.Execution;
+using GraphQL.Types;
+using GraphQLParser.AST;
+
+namespace GraphQL.Validation;
+
+/// <summary>
+/// Represents the context for validating field arguments.
+/// </summary>
+public struct FieldArgumentsValidationContext
```

Comment:
All of the properties and methods here are convenience methods on top of `ValidationContext`.  Since the validation rule already is holding a reference to the field AST and field definition, passing them through the struct is quite efficient.  Also it is much more preferred to use the `Arguments`, `Directives`, and `GetArgument` methods over dealing with `ValidationContext` directly.

Reply:
ReportError is not mapped since exceptions are automatically wrapped by the validation rule, the preferred way of reporting a validation error.